src/uwtools/drivers/driver.py

```diff
@@ -263,6 +260,20 @@ def __init__(
         self._cycle = cycle
         self._leadtime = leadtime
 
+    @property
+    def cycle(self):
+        """
+        The cycle.
+        """
+        return self._cycle
+
+    @property
+    def leadtime(self):
+        """
+        The leadtime.
+        """
+        return self._leadtime
```

Comment:
Here are new read-only properties for the `cycle` and `leadtime` attributes. One or both of them are added to driver classes where appropriate.